pkg/generators/golang/json_generator.go

```diff
@@ -979,10 +979,14 @@ func (g *JSONSupportGenerator) generateListMethodSource(method *concepts.Method)
 					{{ end }}
 				{{ end }}
 				case "items":
-					{{ generateReadValue "items" .Items.Type false }}
+				{{ generateReadValue "items" .Items.Type false }}
+				{{ if and .Items.Type.IsList .Type.Element .Items.Type.Element.IsScalar }}
```

Comment:
I am not 100% sure, but I think that the unlike the "&&" operator in Go the "and" template function isn't lazy: it will evaluate all the expressions and then compute the logical "and". That means that when the type isn't a list the ".Items.Type.Element.IsScalar" will result in a nil pointer error because ".Items.Type.Element" will be "nil". You can work around that by using nested "if" statements. Or else you can modify the implementation of the "IsScalar" function so that it returns "false" when the receiver is nil. As these functions are intended to be easy to use in templates I'd say that the latter is better.

Reply:
I'll modify the implementation of IsScalar. But interestingly when I was testing it locally the test passed I think that the templating changed in some newer golang version as I'm using the go1.21.3

Reply:
The metamodel uses 1.17 so we should bump to newer version to match other ocm projects

Reply:
IsScalar updated and CI passed.
@jhernand could you ptal again?